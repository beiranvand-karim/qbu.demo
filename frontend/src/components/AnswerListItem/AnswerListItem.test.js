/* global describe, it, jest, expect*/
import {shallow} from "enzyme"
import {AnswerListItem} from "./"
import React from "react"
import renderer from "react-test-renderer"
import {MemoryRouter} from "react-router-dom"

describe('<AnswerListItem />', () => {
   const answer = {
      "id": 1,
      "userId": 7,
      "selectedOption": 2,
      "questionId": 25,
      "option": "http://localhost:8282/question/25/2.png",
      "questionTitle": "this question title"
   };
   it('should render without crashing', () => {
      shallow(<AnswerListItem {...answer} />)
   });
   it('should matches the snapshot', () => {
      const tree = renderer.create(<MemoryRouter><AnswerListItem {...answer} /></MemoryRouter>).toJSON();
      expect(tree).toMatchSnapshot()
   });
});