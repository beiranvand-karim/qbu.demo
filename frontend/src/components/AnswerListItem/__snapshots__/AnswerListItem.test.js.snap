// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnswerListItem /> should matches the snapshot 1`] = `
<div>
  <div
    className="alert alert-primary fade"
    role="alert"
  >
    <h2>
      question title: 
      this question title
    </h2>
    <div>
      <strong>
        you chose: 
      </strong>
      <img
        alt=""
        src="http://localhost:8282/question/25/2.png"
      />
    </div>
  </div>
</div>
`;
